pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -93,108 +69,53 @@ func (h *appHandler) handleErr(err error) (ctrl.Result, error) {
 	}, nil
 }
 
-func (h *appHandler) apply(ctx context.Context, appRev *v1beta1.ApplicationRevision, ac *v1alpha2.ApplicationConfiguration, comps []*v1alpha2.Component, policies []*unstructured.Unstructured) error {
-
-	if h.app.Spec.Workflow != nil || ac.Annotations[oam.AnnotationAppRevisionOnly] == "true" {
-		h.FinalizeAppRevision(appRev, ac, comps)
-		err := h.createOrUpdateAppRevision(ctx, appRev)
-		if err != nil {
-			return err
-		}
-		return h.createResourcesConfigMap(ctx, appRev, ac, comps, policies)
+func (h *appHandler) applyAppManifests(ctx context.Context, comps []*types.ComponentManifest, policies []*unstructured.Unstructured) error {
+	appRev := h.currentAppRev
+	if h.app.Spec.Workflow != nil || appRev.Annotations[oam.AnnotationAppRevisionOnly] == "true" {
+		return h.createResourcesConfigMap(ctx, appRev, comps, policies)
+	}
+	if appWillRollout(h.app) {
+		return nil
 	}
 
-	owners := []metav1.OwnerReference{{
-		APIVersion: v1beta1.SchemeGroupVersion.String(),
-		Kind:       v1beta1.ApplicationKind,
-		Name:       h.app.Name,
-		UID:        h.app.UID,
-		Controller: pointer.BoolPtr(true),
-	}}
-	ac.SetOwnerReferences(owners)
-	var err error
+	var latestTracker *v1beta1.ResourceTracker
+	if h.app.Status.LatestRevision != nil {
+		latestTracker = &v1beta1.ResourceTracker{}
+		latestTracker.SetName(dispatch.ConstructResourceTrackerName(h.app.Status.LatestRevision.Name, h.app.Namespace))
+	}
+	// only do GC when ALL resources are dispatched successfully
+	// so skip GC while dispatching addon resources
```

Comment:
why addon resources should skip GC?

Reply:
If not skip GC, GC will delete the previous resource tracker once apply addons successfully. If previous resource tracker is deleted, applying other resources later will fail.